index.html

```diff
@@ -5705,6 +5717,62 @@ <h3>Complex Language Markup</h3>
 
     </section>
 
+    <section class="appendix normative">
+      <h2>Verification Algorithm</h2>
+
+      <p>
+This section contains an algorithm that <a>conforming verifier processors</a>
+can run when verifying a <a>verifiable credential</a> or a <a>verifiable
+presentation</a>. Any algorithm can be used that produces output that
+is equivalent to the algorithm defined in this section. This algorithm takes a
+byte stream (<var>inputBytes</var>) and a media type (<var>inputMediaType</var>)
+as inputs, and produces an object (<var>verified</var>) that contains a
+verification status (<var>verified</var>.<var>status</var>),
+a verified document (<var>verified</var>.<var>document</var>),
+a media type (<var>verified</var>.<var>mediaType</var>), zero or more
+optional warnings (<var>verified</var>.<var>warnings</var>), and
+zero or more optional errors (<var>verified</var>.<var>errors</var>).
+      </p>
+
+      <ol class="algorithm">
+        <li>
+Set <var>verified</var> to an empty object.
+        </li>
+        <li>
+Verify that the securing mechanism in use on <var>inputBytes</var> has not been
+tampered with, using <var>inputMediaType</var> as additional input to the
+algorithm. See Section <a href="#securing-verifiable-credentials"></a>
+for details.
+        </li>
+        <li>
+If the verification of the securing mechanism is successful, set
+<var>verified</var>.<var>status</var> to `true`,
+<var>verified</var>.<var>document</var> to the information that was secured,
+<var>verified</var>.<var>mediaType</var> to the corresponding media type,
+and optionally <var>verified</var>.<var>warnings</var> to any warning
+objects returned by the verification process for the securing mechanism.
+          <ol class="algorithm">
+            <li>
+Check whether <var>verified</var>.<var>document</var> is a
+<a>conforming document</a>. If not, optionally add appropriate warnings
```

Comment:
We've written the spec such that all relevant normative statements include "MUST". 

Your concern here seems to be less about the algorithm and more about normative language elsewhere in the specification, correct? So, what we could do is raise a new issue with your list above (to start). Would that work for you? 

IOW, would you be fine w/ us dealing w/ the other normative statements in another issue and commenting on the changes you'd like to be made to the text above (presuming we'll address that other issue in time)?

Reply:
Yes, and to be clear, I'm always happy to break up work into multiple PRs and let some of it land before all of it's done.

I'm worried about using the declarative style of "A [conforming document](https://w3c.github.io/vc-data-model/#dfn-conforming-document) is any concrete expression of the data model that complies with the normative statements in this specification," rather than the algorithmic style of "follow these steps to check if a document is conforming." Based on the web platform's experience with HTML, I think the declarative style is more likely to lead to differences in implementation and that it'll be harder to check for privacy or security problems. However, since this algorithm makes it optional to take any action based on non-conformance, I'm not firmly opposed to the declarative style in this particular call.

Reply:
Yes, the first draft of this PR used the algorithmic style, which duplicated the normative statements throughout the document. As you know, that leads to people complaining about duplicating normative statements AND often leads to sync problems between the two duplicated normative statements throughout the specification. We could split the difference and say something like: "Ensure that the `foo` field conforms to the normative statements in Section 2.5 Foo, and if not, raise an error." -- it points out more clearly what fields are expected to be checked during verification w/o duplicating normative text. 

If you're ok w/ that (I know it's not exactly what you were asking for), I can take a shot at some language in that direction.

Reply:
To be clear, you don't need to make me happy with the declarative style in places like this where enforcement is optional. (So, yes, I'm ok with your suggestion.) But I do think it doesn't serve you well, for 2 reasons, both based on the fact that you're fundamentally trying to constrain two different pieces of software with this language.

1. Spec authors tend to only think about one audience at a time, so language with two audiences (here, the issuer and verifier) tends to lose its applicability to one of the audiences over time. Here, since the text is mostly directed to the issuers, it has lost the detail that verifiers would need. Some specs, like https://httpwg.org/specs/rfc9110.html#rfc.section.4.2.1, have successfully interleaved their two audiences in the same text, but it's unusual.
2. Sometimes you really want to put different constraints on issuers vs verifiers. For example, in https://github.com/w3c/vc-data-model/pull/1349/files#r1391380302, I think it's very reasonable to say that the issuer "MUST identify the issuer in a globally unambiguous way" even though it's not reasonable for a verifier to check that. It'd be good to use a specification style that allows you to separate the two. (That said, I think it's also ok to only constrain the consumer, as web specs usually do.)

Reply:
I have made a first attempt at expanding the algorithm to be more precise here: chrome://settings/content/all